fnet/cli/predict.py

```diff
@@ -200,89 +210,66 @@ def save_args_as_json(args: argparse.Namespace) -> None:
-
-    Parameters
-    ----------
-    path_pred_csv
-        Path to prediction results CSV.
-    metric
-        Name of metric.
 
-    Returns
-    -------
-    Dict[str, float]
-        Mean metric score for each model.
-
-    """
-    df = pd.read_csv(path_pred_csv)
-    cols_keep = [c for c in df.columns if c.startswith(metric)]
-    return (
-        df.filter(cols_keep)
-        .mean(axis=0)
-        .to_dict()
-    )
+def load_from_json(args: argparse.Namespace) -> None:
+    """Loads arguments from if a json is specified."""
+    if args.json is None:
+        return
+    with args.json.open(mode='r') as fi:
+        predict_options = json.load(fi)
+    args.__dict__.update(predict_options)
 
 
 def add_parser_arguments(parser) -> None:
     """Add training script arguments to parser."""
-    parser.add_argument('path_model_dir', nargs='+', help='path(s) to model directory')
     parser.add_argument('--dataset', help='dataset name')
     parser.add_argument('--dataset_kwargs', type=json.loads, default={}, help='dataset kwargs')
-    parser.add_argument('--gpu_ids', nargs='+', type=int, default=[0], help='GPU ID')
+    parser.add_argument('--gpu_ids', type=int, default=0, help='GPU ID')
     parser.add_argument('--idx_sel', nargs='+', type=int, help='specify dataset indices')
+    parser.add_argument('--json', type=Path, help='path to prediction options json')
     parser.add_argument('--metric', default='fnet.metrics.corr_coef', help='evaluation metric')
     parser.add_argument('--n_images', type=int, default=-1, help='max number of images to test')
     parser.add_argument('--no_prediction', action='store_true', help='set to not save predicted image')
     parser.add_argument('--no_signal', action='store_true', help='set to not save signal image')
     parser.add_argument('--no_target', action='store_true', help='set to not save target image')
+    parser.add_argument('--path_model_dir', nargs='+', help='path(s) to model directory')
     parser.add_argument('--path_save_dir', default='predictions', help='path to output root directory')
     parser.add_argument('--path_tif', help='path(s) to input tif(s)')
 
 
-def main(args: Optional[argparse.Namespace] = None) -> Dict[str, float]:
-    """Predicts using model.
-
-    Parameters
-    ----------
-    args
-        Predict arguments.
-
-    Returns
-    -------
-    Dict[str, float]
-        Mean metric score for each model.
-
-    """
+def main(args: Optional[argparse.Namespace] = None) -> None:
+    """Predicts using model."""
     if args is None:
         parser = argparse.ArgumentParser()
         add_parser_arguments(parser)
         args = parser.parse_args()
-    path_pred_csv = Path(args.path_save_dir, 'predictions.csv')
-    if path_pred_csv.exists():
-        LOGGER.info(f'Using existing prediction results: {path_pred_csv}')
-        return aggregate_results(path_pred_csv, args.metric)
+    if (args.json is not None) and (not args.json.exists()):
```

Comment:
Can be shortened to:

```suggestion
    if args.json and not args.json.exists():
```